### PR TITLE
Update JSDoc

### PR DESCRIPTION
**Pull request by oandregal**
Update JSDoc comment for `concat` function.

### CODE REVIEW

**Comment by iseulde** [comment]
Is this part of a systematic check of all documentation for RichText?

File: packages/rich-text/src/concat.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { normaliseFormats } from './normalise-formats';
  * Combine all Rich Text values into one. This is similar to
  * `String.prototype.concat`.
  *
- * @param {...[object]} values An array of all values to combine.
+ * @param {...Object} values Objects to combine.
```
Comment (iseulde) [inline packages/rich-text/src/concat.js]: I thought I'd seen the removed syntax around... The one you're suggesting seems correct but it's a bit weird to me. `values` is an array, not an object?

Comment (nosolosw) [reply packages/rich-text/src/concat.js]: I'm working on [generating documentation (markdown, html) from the JSDoc comments](https://github.com/WordPress/gutenberg/pull/13329) and this one didn't pass my tests. I didn't find this way of param type declaration in the [standard](http://usejsdoc.org/tags-type.html), either.

I thought it took objects as input? As in `concat( object1, object2, object3 );`. That is what is implied in the tests. `concat` converts that input into an array, though:

`export function concat( ...values ) { }`

Comment (aduth) [reply packages/rich-text/src/concat.js]: Changes as proposed seem correct to me. If it receives variadic arguments, it should be a `...` prefix on the type expected.

From [your link](http://usejsdoc.org/tags-type.html), see under "Variable number of that type". The main difference is ES2015+ adds syntax support for the spread argument.

Related:

- https://github.com/jsdoc3/jsdoc/issues/1074
- https://github.com/jsdoc3/jsdoc3.github.com/pull/183

**Review by ellatrix — APPROVED** [review]

**Pull request merged**